extensions/amp-story-interactive/amp-story-interactive.md

```diff
@@ -1,10 +1,310 @@
+
+## Aggregate data source
+
+All selectable interactive components (not results) show the percentage of users that selected each option. This data is aggregated on a backend specified with the `endpoint` attribute.
+To fetch the data for an interactive component, the necessary fields are:
+
+- `interactiveId`: the `base64encode(CANONICAL_URL) + "+" + element.id`
+- `interactiveType`: enum from [amp-story-interactive-abstract:48](https://github.com/ampproject/amphtml/blob/3a86226fe428ce72adb67cffe2dd2f1fae278a35/extensions/amp-story-interactive/1.0/amp-story-interactive-abstract.js#L48)
+- `endpoint`: the attribute `element.getAttribute("endpoint")`
+- `ampId`: client ID that identifies the session, optional
+
+Then, the requests and responses are:
+
+[sourcecode:js]
+// Getting the votes for an interactive.
+// If no client param, selected will always be false
+// (can be used to display the results on a dashboard).
+
+GET_URL = "{endpoint}/{interactiveId}?type={interactiveType}&client={ampId}"
+
+Response: {
+"options": [
+{"index": 0, "selected": false, "count": 0},
+{"index": 1, "selected": false, "count": 5},
+{"index": 2, "selected": false, "count": 7},
+{"index": 3, "selected": false, "count": 2}
+]
+}
+[/sourcecode]
+
+[sourcecode:js]
+// Posting the vote for an interactive. Client param is required.
+
+POST_URL = "{endpoint}/{interactiveId}:vote?type={interactiveType}&client={ampId}"
+POST_BODY = {'option_selected': 1}
+
+Response: No response necessary
+[/sourcecode]
+
+Backends need to be specified on the necessary components (binary-poll, poll, quiz), and can be deployed by publishers, tools or others. Available free-to-use backends are:
+
+- Google hosted: _coming soon_
+
+[tip type="note"]
+Before setting up a backend, consider if the already existing backends satisfy your requirements.
+[/tip]
+
+## Styling
+
+If you want to see all the theming options in action, check out the [example story](/documentation/examples/components/amp-story-interactive-poll/story#page=title-themes)
+
+### CSS Variables
+
+All interactive components support similar styling, which is controlled through CSS variables and attributes.
+CSS variables
+Some variables can be overridden through assigning a class to the component.
+
+- `--interactive-accent-color`: The accent color of the component. If no prompt-background is specified, it will use that for the prompt background as well.
+- `--interactive-prompt-text-color`: Color of the top text. Only used on selectable components if a
```

Comment:
Used that naming on the top as well, it refers to the elements that have options that can be selected. Quizzes, polls and binary polls can be selected, but results are not selected (they are inferred). Since "components" is not the right word here, I'll use elements